src/Solver/Interoperability.cpp

```diff
@@ -471,59 +483,123 @@ void seissol::Interoperability::setupFSRMPointSources( double const* momentTenso
+  //There are only some valid combinations of material properties
+  // elastic materials
+  // viscoelastic materials
+  // elastoplastic materials
+  // viscoplastic materials
+  // anisotropic elastic materials
   
-  if (anelasticity == 1) {
-    parameterDB.addParameter("Qp",  materialVal + nElements*3);
-    parameterDB.addParameter("Qs",  materialVal + nElements*4);
-  }
 
-  if (plasticity == 1) {
-    parameterDB.addParameter("bulkFriction", bulkFriction);
-    parameterDB.addParameter("plastCo",      plastCo);
-    parameterDB.addParameter("s_xx",         iniStress+0, 6);
-    parameterDB.addParameter("s_yy",         iniStress+1, 6);
-    parameterDB.addParameter("s_zz",         iniStress+2, 6);
-    parameterDB.addParameter("s_xy",         iniStress+3, 6);
-    parameterDB.addParameter("s_yz",         iniStress+4, 6);
-    parameterDB.addParameter("s_xz",         iniStress+5, 6);
-  }
-  
+  //first initialize the (visco-)elastic part
+  auto nElements = seissol::SeisSol::main.meshReader().getElements().size();
   seissol::initializers::ElementBarycentreGenerator queryGen(seissol::SeisSol::main.meshReader());
-  parameterDB.evaluateModel(std::string(materialFileName), queryGen);
+  if (anisotropy) { 
+    if(anelasticity || plasticity) {
+      logError() << "Anisotropy can not be combined with anelasticity or plasticity";
+    }
+    auto materials = std::vector<seissol::model::AnisotropicMaterial>(nElements);
+    seissol::initializers::MaterialParameterDB<seissol::model::AnisotropicMaterial> parameterDB;
+    parameterDB.setMaterialVector(&materials);
+    parameterDB.evaluateModel(std::string(materialFileName), queryGen);
+    for (unsigned int i = 0; i < nElements; i++) {
+      materialVal[i] = materials[i].rho;
+      materialVal[nElements + i] = materials[i].c11;
+      materialVal[2*nElements + i] = materials[i].c12;
+      materialVal[3*nElements + i] = materials[i].c13;
+      materialVal[4*nElements + i] = materials[i].c14;
+      materialVal[5*nElements + i] = materials[i].c15;
+      materialVal[6*nElements + i] = materials[i].c16;
+      materialVal[7*nElements + i] = materials[i].c22;
+      materialVal[8*nElements + i] = materials[i].c23;
+      materialVal[9*nElements + i] = materials[i].c24;
+      materialVal[10*nElements + i] = materials[i].c25;
+      materialVal[11*nElements + i] = materials[i].c26;
+      materialVal[12*nElements + i] = materials[i].c33;
+      materialVal[13*nElements + i] = materials[i].c34;
+      materialVal[14*nElements + i] = materials[i].c35;
+      materialVal[15*nElements + i] = materials[i].c36;
+      materialVal[16*nElements + i] = materials[i].c44;
+      materialVal[17*nElements + i] = materials[i].c45;
+      materialVal[18*nElements + i] = materials[i].c46;
+      materialVal[19*nElements + i] = materials[i].c55;
+      materialVal[20*nElements + i] = materials[i].c56;
+      materialVal[21*nElements + i] = materials[i].c66;
```

Comment:
Does the fortran part actually need the material values anymore?

Reply:
Indeed it does.
Material values are read into the Fortran part in src/Physics/ini_model.f90:130
They are written back to CPP in src/Initializer/dg_setup.f90:556
It seems that the material values are not needed in the rest of the Fortran part. I will have a look into how to remove this.

Reply:
It's quite a pain to remove the material from the Fortran part. I have some ideas about how to do this, but I think this is beyond something that can easily be done within this PR.